src/EGraphs/egraph.jl

```diff
@@ -291,7 +273,7 @@ function add!(g::EGraph{ExpressionType,Analysis}, n::VecExpr, should_copy::Bool)
   g.memo[n] = id
 
   add_class_by_op(g, n, id)
-  eclass = EClass{Analysis}(id, VecExpr[n], Pair{VecExpr,Id}[], make(g, n))
+  eclass = EClass{Analysis}(id, VecExpr[copy(n)], Pair{VecExpr,Id}[], make(g, n))
```

Comment:
if you see above, there's a `should_copy` argument. Is this copy required? Why?

Reply:
Corresponding code from egg:

```rust
        let class = EClass {
            id,
            nodes: vec![enode.clone()],
            data: N::make(self, &original),
            parents: Default::default(),
        };
```
https://github.com/egraphs-good/egg/blob/b3a53e9e575dc67ce1e8320e4e488f2873a67482/src/egraph.rs#L1076C1-L1081C11